Automattic/simplenote-macos
# Checklists: Undo Exceptions. Mark I

### Description:
This is the first part of a few PR(s) aiming at:

A. Encapsulating all the things Checklists
B. Refactoring the main mechanism, so that the UndoManager doesn't throw exceptions
C. Testing all the things

### Details:
- Added new Unit Testing Target. **Seriously**
- Updated the main Podfile so that 10.13 is the deployment target (matching the main project)
- Encapsulated the **NSAttributedString** to **Markdown** converter into an actual class
- Added Unit Test for the above!
- Updated CircleCI so that Unit Tests are ... effectively executed all over!
- And minor touches here and there (dead code deletion // overall cleanup)

cc @aerych (Thank you Eric!!!)
Ref. #467

### Test
1. Add a new note
2. Insert a checklist
3. Add several items (and swizzle the checked / unchecked state)
4. Relaunch the app

- [x] Verify that the note looks exactly the same way it was before relaunching
- [x] Verify the (new!) Unit Tests are green

### Release
These changes do not require release notes.
